### PR TITLE
Run scheduled benchmark test on self-hosted runner

### PR DESCRIPTION
**Pull request by gyohuangxin**
**Description** 

This PR fixes https://github.com/layer5io/meshery-smp-action/issues/38, implements running scheduled benchmark test on self-hosted runner.

There are two method Scheduled benchmarking test can be triggered:
**Manual test**: we can trigger the test by giving specific test configurations (load-test, soak-test), then the matrix will be tested.
**Scheduled test**: the matrix test will run everyday.

There are three steps for scheduled benchmarking test like configurable benchmarking test: 
1. Create CNCF CIL machines and register them as github self-hosted runners
2. Install Kubernetes, Service mesh and meshery. Then run benchmarking tests.
3. Destroy machines and delete self-hosted runners. 
Workflows:
image

Github self-hosted runners:
image
CNCF CIL machines:
image

**Notes for Reviewers**

**[Signed commits](../CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by gyohuangxin** [comment]
This PR also fixed https://github.com/layer5io/meshery-smp-action/issues/21, the names of  test profile, machines, and runners will not contain `.yaml`

**Comment by gyohuangxin** [comment]
@hershd23 @navendu-pottekkat  @leecalcote  Can you review and give any comments? Thanks.

File: .github/workflows/scheduled-benchmarks-self-hosted.yaml [hunk 1]
```diff
@@ -0,0 +1,277 @@
...
+          curl -X DELETE -H 'Authorization: token ${{ secrets.PAT }}' -H "Accept: application/vnd.github.v3+json" https://api.github.com/repos/${{github.repository}}/actions/runners/$runner_id
+        shell: bash
+
+  # Scheduled Benchmark Test
+  start-runners-scheduled:
+    name: Start self-hosted CNCF CIL runners for scheduled test
+    runs-on: ubuntu-latest
+    if: ${{ github.event_name == 'schedule' }}
+    strategy:
+      fail-fast: false
```
Comment (hershd23) [inline .github/workflows/scheduled-benchmarks-self-hosted.yaml:154]: Any reason for this

**Review by hershd23 — COMMENTED** [review]
Great work @gyohuangxin. I have made one comment.

I have one more thing to ask, the manual and scheduled runs reuse almost all the code, what is the reason for having another copy entirely for scheduled? Is there any difference in logic that I am unable to see?

File: .github/workflows/scheduled-benchmarks-self-hosted.yaml (hunk 1 above)
Comment (gyohuangxin) [reply .github/workflows/scheduled-benchmarks-self-hosted.yaml:154]: When jobs.<job_id>.strategy.fail-fast is set to true, GitHub cancels all in-progress jobs if any matrix job fails. Default: true
https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstrategyfail-fast
Obviously, all tasks run in parallel and should not affect each other.

**Comment by gyohuangxin** [comment]
> Great work @gyohuangxin. I have made one comment.
> 
> I have one more thing to ask, the manual and scheduled runs reuse almost all the code, what is the reason for having another copy entirely for scheduled? Is there any difference in logic that I am unable to see?

Thanks for your comments, the matrixs are different between manual and scheduled runs. And I think it's hard to reuse codes from anyone of them because the event types are also different.

**Review by hershd23 — APPROVED** [review]
LGTM

**Comment by leecalcote** [comment]
Excellent. 😄

**Pull request merged**